### PR TITLE
[nataliia romanchuk/key_words]  tests

### PR DESCRIPTION
**Pull request by Vivanatalya**
added tests for testing key words

### CODE REVIEW

File: tests/nataliia_romanchuk/test_key_words.py [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+
+
+def test_and():
+    assert (1 == 1) and ('1' == '1')
+    assert not ((1 == 1) and ('1' == 1))
+
+
+def test_or():
+    assert (1 == 1) or (1 == '1')
+    assert not (1 == 1) or ('1' == '1')
```
Comment (kudinov-fedor) [inline tests/nataliia_romanchuk/test_key_words.py:18]: this test should fail,
do not write failing tests

Comment (kudinov-fedor) [inline tests/nataliia_romanchuk/test_key_words.py:18]: or  and

try cases when they return objects e,g,

("abc" and "def") == "def"
("" and "def") == ""

File: tests/nataliia_romanchuk/test_key_words.py [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+    assert 'sausage' not in 'cat'
+
+
+def test_if_else():
+    x = 1
+    if x > 0:
+        result = True
+    else:
+        result = False
+    assert result == True
```
Comment (kudinov-fedor) [inline tests/nataliia_romanchuk/test_key_words.py:49]:
```
def test_if_else():
    x = 1
    if x > 0:
        pass
    else:
        raise AssertionError

```

File: tests/nataliia_romanchuk/test_key_words.py [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+    x = 0
+    while x < 5:
+        x += 1
+    assert x == 5
+
+
+def test_list_comprehension():
+    numbers = [1, 11, 111]
+    squares = [x ** 2 for x in numbers]
+    assert squares == [1, 11**2, 111**2]
```
Comment (kudinov-fedor) [inline tests/nataliia_romanchuk/test_key_words.py:70]: spaces around operators

**Review by kudinov-fedor — APPROVED** [review]

**Comment by kudinov-fedor** [comment]
try also conditional expressions

File: tests/nataliia_romanchuk/test_key_words.py (hunk 1 above)
Comment (Vivanatalya) [reply tests/nataliia_romanchuk/test_key_words.py:18]: fixed

File: tests/nataliia_romanchuk/test_key_words.py (hunk 2 above)
Comment (Vivanatalya) [reply tests/nataliia_romanchuk/test_key_words.py]: fixed

File: tests/nataliia_romanchuk/test_key_words.py (hunk 3 above)
Comment (Vivanatalya) [reply tests/nataliia_romanchuk/test_key_words.py]: fixed

File: tests/nataliia_romanchuk/test_key_words.py (hunk 1 above)
Comment (Vivanatalya) [reply tests/nataliia_romanchuk/test_key_words.py:18]: added

File: tests/nataliia_romanchuk/test_key_words.py [hunk 4]
```diff
@@ -0,0 +1,106 @@
...
+
+def test_and():
+    assert (1 == 1) and ('1' == '1')
+    assert not ((1 == 1) and ('1' == 1))
+
+
+def test_or():
+    assert (1 == 1) or (1 == '1')
+    assert not (1 == 1) or ('1' == '1')
+    assert "abc" and "def" == "def"
```
Comment (fkudinov-mn) [inline tests/nataliia_romanchuk/test_key_words.py:19]: i suppose execution order is not what you expect

it is:
```
"abc" and ("def" == "def")
```

File: tests/nataliia_romanchuk/test_key_words.py [hunk 5]
```diff
@@ -0,0 +1,106 @@
...
+    numbers = [1, 11, 111]
+    squares = [x ** 2 for x in numbers]
+    assert squares == [1, 11 ** 2, 111 ** 2]
+
+
+def test_dict_comprehension():
+    keys = range(1, 4)
+    values = [1, 11, 111]
+    my_dict = {key: value for key, value in zip(keys, values)}
+    square_dict = {key: value ** 2 for key, value in my_dict.items()}
```
Comment (fkudinov-mn) [inline tests/nataliia_romanchuk/test_key_words.py:92]: this one better to sepparate test

File: tests/nataliia_romanchuk/test_key_words.py (hunk 4 above)
Comment (Vivanatalya) [reply tests/nataliia_romanchuk/test_key_words.py]: fixed

**Pull request merged**